src/gui/mzroll/tabledockwidget.cpp

```diff
@@ -1788,6 +1843,43 @@ void TableDockWidget::contextMenuEvent(QContextMenuEvent *event) {
   QAction *z4 = menu.addAction("Delete all groups from this table");
   connect(z4, SIGNAL(triggered()), SLOT(deleteAll()));
 
+  QList<QPointer<TableDockWidget>> peaksTableList = _mainwindow->getPeakTableList();
+  int n = peaksTableList.count();
+
+  QMenu* subMenu = menu.addMenu(tr("Move to Table"));
+  
+  QMap<QAction*, QString> actionsAndTables;
+  QString tableName = _mainwindow->bookmarkedPeaks->titlePeakTable->text();
+  if(this->titlePeakTable->text() != tableName ) {
+    QAction* z5 = subMenu->addAction(tableName);
+    actionsAndTables[z5] = tableName;
+  }
+  
+  for (int i = 0; i < n; i++) {
+    tableName = peaksTableList[i]->titlePeakTable->text();
+    if(this->titlePeakTable->text() != tableName){
+      QAction* z5 = subMenu->addAction(tableName);
+      actionsAndTables[z5] = tableName;
+    }
+  }
+
+  QList<shared_ptr<PeakGroup>> selected = getSelectedGroups();
+
+  QMapIterator<QAction*, QString> iter(actionsAndTables);
+  while (iter.hasNext()) {
+    iter.next();
+    QAction* action = iter.key();
+    QString tableName = iter.value();
+    connect(action,
+            &QAction::triggered,
+            this,
+            [this, tableName]() { this->moveSelectedRows(tableName); });
+    if (selected.count() == 0) {
+      // disable action not relevant when nothing is selected
+      action->setEnabled(false);
+    }
```

Comment:
I know we went over this before you opened a PR, but I think it makes more sense to simply disable the parent menu item instead of disabling each child action item. So "Move to another table…" will itself be grayed when nothing is selected in that table.

Reply:
yeah right it makes more sense